core/sql/exp/ExpLOBaccess.cpp

```diff
@@ -1502,7 +1507,50 @@ Ex_Lob_Error ExLob::allocateDesc(ULng32 size, Int64 &descNum, Int64 &dataOffset,
     char logBuf[4096];
     lobDebugInfo("In ExLob::allocateDesc",0,__LINE__,lobTrace_);
     Int32 openFlags = O_RDONLY ;   
-    
+    if (size == 0) //we are trying to empty this lob.
+      {
+        //rename lob datafile
+        char * saveLobDataFile = new(getLobGlobalHeap()) char[MAX_LOB_FILE_NAME_LEN+6];
```

Comment:
Should this be MAX_LOB_FILE_NAME_LEN+6+1 (to allow for "s_save" and for a trailing null)?